lib/kino_slack/message_cell.ex

```diff
@@ -78,7 +79,24 @@ defmodule KinoSlack.MessageCell do
     end
   end
 
-  def all_fields_filled?(attrs, keys) do
+  defp interpolate(message) do
+    ~r/(?<before_interpolation>){{[^}]+}}(?<after_interpolation>)/
```

Comment:
The interpolated content doesn't have to be a variable, it could be regular code with list `]` or map `}` inside.

Also we can do this with a single pass on the string looking for `{{`, which is more predictable performance-wise :) See [this clause](https://github.com/livebook-dev/kino_db/blob/c0f6f4a15c7b3b515f0de9be02bd776c997fa35e/lib/kino_db/sql_cell.ex#L262-L269), except in this case we want to build AST for the whole interpolated string.

Reply:
@jonatanklosko 

Changed some code, trying to follow your suggestions.

Please take a look. 😄